src/react_native/core.cljs

```diff
@@ -199,6 +199,15 @@
   [handler deps]
   (react/useMemo handler (get-js-deps deps)))
 
+(defn delay-render
+  [content]
+  (let [[render? set-render] (use-state false)]
+    (use-mount
+     (fn []
+       (js/setTimeout #(set-render true) 0)))
+    (when render?
+      content)))
+
```

Comment:
Hey @ilmotta, I have seen the discussion on discord about this. I checked out the builds in this PR and compared to `develop` on iOS and Android devices. In my experience both of develop and your PR are very and equally smooth. But I see in the videos you posted it performs better on your Android device (but seems not as smooth as on my device). I can make a small suggestion here, have you tried using `reagent/next-tick` instead of `js/setTimeout 0` it can perform slightly better as it places the function on a micro-task queue rather than a macro-task queue like `setTimeout`

```
(defn delay-render
  [content]
  (let [[render? set-render] (use-state false)]
    (use-mount
     (fn []
       (reagent/next-tick #(set-render true))))
    (when render?
      content)))
  ```
  
Here is an Android build that uses that `https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240508-131932-35375227960bc897cdeb51c9d9a5279139ca9174-pr19941-universal.apk`

Reply:
Hi @OmarBasem, thanks a lot for reviewing the build and code. I expected iOS would perform nearly the same because things are much smoother over there. Now that you confirmed this, I guess we can say this PR is really targeting Android.

I remember when `delay-render` was first implemented there was a comment about using `next-tick` (maybe @ulisesmac said that), but I don't remember why we ended up using `setTimeout` (maybe because it's well known among devs and the performance was the same at the end of the day).

I'll check out with `next-tick` and see what comes out of it! Thanks

Reply:
@OmarBasem, I tried with `next-tick`. It works the same on average. Here's how I compared them, which I think is not super scientific, but quite practical from the user's perspective.

1. I enabled show taps and recorded the current and new implementation opening and closing the Settings screen many times.
2. After the first video frame shows the white circle (press) I count the number for frames in the video until the Settings screen is rendered and animated at the top. There's a tiny margin for error counting like this, but should be precise enough.

The results (ignore absolute frame numbers, just how close they all are):

```
next-tick: 12 frames, 11 frames, 9 frames
setTimeout: 10 frames, 8 frames, 13 frames, 10 frames
```

So, I would say they perform essentially the same for this particular component, but I agree `next-tick` can theoretically be better, even if no practical advantage could be measured for the Settings screen. Even if the number of frames to render is the same, maybe `next-tick` could consume less resources? But anyway, I'll use next-tick because it's cleaner and gives the same effect.

Reply:
I think `next-tick` could communicate the intent more clearly, because it seems we're trying to defer the rendering to relieve the cpu during some animations (?).

Reply:
> But anyway, I'll use next-tick because it's cleaner and gives the same effect.

> I think next-tick could communicate the intent more clearly

@ilmotta @seanstrom I agree, when I find both from `next-tick` and `setTimeout 0` having the same effect practically I use `next-tick` as I think it communicates the intent better

Reply:
should we make this solution android specific so? i.e split it by platform? 🤔

Reply:
@J-Son89 the solution introduced in Icaro's PR is to use a hook instead of hiccup for `delay-render`

As for `next-tick`, as discussed above the performance in this scenario is the same but we settled for it over `setTimeout 0` because it is a little cleaner. It is not platform specific.

Reply:
Cool cool, it's just that in your initial comment you said this was not as smooth for you on iOS.

Anyway I guess it fixes the bigger issue which is a bad android performance so that's great 🚀